core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3072,15 +3072,14 @@ private[spark] object Utils
    */
   lazy val isG1GC: Boolean = {
     Try {
+      // SPARK-48505: If the initialization probe of `HotSpotDiagnosticMXBean` is successful,
+      // the API of `HotSpotDiagnosticMXBean` can be used directly in subsequent operations,
+      // instead of invoking it through reflection.
       val clazz = Utils.classForName("com.sun.management.HotSpotDiagnosticMXBean")
-        .asInstanceOf[Class[_ <: PlatformManagedObject]]
-      val vmOptionClazz = Utils.classForName("com.sun.management.VMOption")
+      import com.sun.management.HotSpotDiagnosticMXBean
```

Comment:
Although the code becomes simpler, I'm not sure this is a better recommended style: `import com.sun.*` in our source code.

There is a similar discussion here.
- https://github.com/apache/spark/pull/43503#discussion_r1671300283

Reply:
Thanks @dongjoon-hyun ,understood, that makes sense. Let's keep things as they are; I will close this pr.